functions/prepared-check-sync/v1.js

```diff
@@ -0,0 +1,93 @@
+const moment = require('moment')
+const R = require('ramda')
+const uuid = require('uuid/v4')
+
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
+const accessArrangementsSqlUtil = require('./access-arrangements-sql-util')
+const preparedCheckSchemaValidator = require('../lib/prepared-check-schema-validator')
+
+const v1 = {}
+
+v1.process = async function process (context, preparedCheckSyncMessage) {
+  const { checkCode } = preparedCheckSyncMessage
+  context.log('prepared-check-sync: message received', checkCode)
+  let preparedCheck
+  try {
+    preparedCheck = await azureStorageHelper.getFromPreparedCheckTableStorage(azureTableService, checkCode, context.log)
+  } catch (error) {
+    context.log.error(`prepared-check-sync: ERROR: unable to fetch preparedCheck from table storage for checkCode [${checkCode}]`)
+    throw error
+  }
+  let pupilAccessArrangements
+  try {
+    pupilAccessArrangements = await accessArrangementsSqlUtil.sqlFindPupilAccessArrangementsByCheckCode(checkCode)
```

Comment:
It seems a pity that the access arrangements are not included in the message payload.  I doubt the lookup is a problem though, but it is another dependency that can go wrong.

Reply:
This would require additional changes down the road on the pupil spa to align but certainly can be explored as an improvement.